test/connectivity/modules/NetworkController/NetworkInterfaceOfflineController.cs

```diff
@@ -23,73 +21,29 @@ public NetworkInterfaceOfflineController(string dockerInterfaceName)
 
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                this.networkCommands = new WindowsNetworkInterfaceCommands();
+                this.underlyingConroller = new WindowsNetworkInterfaceOfflineController();
             }
             else
             {
-                this.networkCommands = new LinuxNetworkInterfaceCommands(this.networkInterfaceName);
+                this.underlyingConroller = new LinuxNetworkInterfaceOfflineController(this.networkInterfaceName);
             }
         }
 
         public string Description => "NetworkInterfaceOffline";
 
         public Task<NetworkStatus> GetStatus(CancellationToken cs)
         {
-            OperationalStatus status = this.GetNetworkInterfaceStatus();
-            switch (status)
-            {
-                case OperationalStatus.Up:
-                    return Task.FromResult(NetworkStatus.Default);
-                case OperationalStatus.Unknown:
-                    return Task.FromResult(NetworkStatus.Unknown);
-                default:
-                    return Task.FromResult(NetworkStatus.Restricted);
-            }
+            return this.underlyingConroller.GetStatus(cs);
         }
 
-        public async Task<bool> AddNetworkControllingRule(CancellationToken cs)
+        public async Task<bool> SetStatus(NetworkStatus status, CancellationToken cs)
```

Comment:
does this method have same implementation in FirewallOfflineController class? if so, can it be refactored for reuse?